### ISSUE TITLE
NUnit2010 and `EqualConstraint`

### ISSUE THREAD

**Issue by JasonBock** [nunit/nunit.analyzers#380]
I have a class that implements `IEquatable<T>` as well as the `==` operator. I want to write a test that uses the constraint model that will test both paths: `x == y` and `x.Equals(y)`. If I do this:
```
Assert.That(x == y, Is.True);
```
I get the NUnit2010 warning, and since I have `TreatWarningsAsErrors` set to `true`, this stops the compiler. I understand why this is being reported, but I'm wondering if `EqualConstraint` could get a `EqualToViaOperator` method. That way I could test both paths in my code.

This may not be possible right now because you can't generically determine if a type implements an operator. This may be possible in C# 10 with [static abstract members in interfaces](https://github.com/dotnet/csharplang/issues/4436) if the feature makes it into that version. 

Seems like the best approach for now would be to use a `#pragma warning disable` around the code.

**Comment by mikkelbu** [comment]
NUnit2010 should not report a warning, only a message with "Information" as severity.

Instead of `#pragma warning disable` couldn't you "remove" the diagnostic, if you extract the result into a variable - e.g.
```csharp
var comparison = x == y;
Assert.That(comparison, Is.True);
```

I can see that it is problematic that the refactoring suggest changing the code into `Assert.That(x, Is.EqualTo(y));` which has a different meaning, but I'm unsure if we can do better. I'll keep this issue open for possible improvements, but it can be that the best we can do (for now) is to mention this in the documentation.

Regarding whether `EqualConstraint` could get a `EqualToViaOperator`, then this should be asked in the main NUnit repository - https://github.com/nunit/nunit. I don't think I've seen an issue like this in the project.

Ps. And thanks for creating this project 👍.

**Comment by JasonBock** [comment]
I keep forgetting that I'm overtly aggressive with analyzer severity settings. I turned all of the NUnit ones to `error` in my .editorconfig :)

Totally understand about if there's not much to do at this point. I'll put an feature issue in the NUnit repo as something to consider if that C# feature is included.

And to be clear....I may have talked to the NUnit team members to start an analyzers project, but I've done no work on the analyzers themselves. Other folks have done the hard work, and they deserve the credit :)
